glib/src/signal.rs

```diff
@@ -146,3 +146,81 @@ pub fn signal_has_handler_pending<T: ObjectType>(
         ))
     }
 }
+
+// rustdoc-stripper-ignore-next
+/// Whether an event was handled or not.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum Propagation {
+    // Event was handled, don't propagate
+    Stop,
+    // Event was not handled, propagate
+    Proceed,
+}
+
+impl Propagation {
+    // rustdoc-stripper-ignore-next
+    /// Returns `true` if this is a `Stop` variant.
+    pub fn is_stop(&self) -> bool {
+        matches!(self, Self::Stop)
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Returns `true` if this is a `Stop` variant.
+    pub fn is_proceed(&self) -> bool {
+        matches!(self, Self::Proceed)
+    }
+}
+
+impl From<bool> for Propagation {
+    fn from(c: bool) -> Self {
+        if c {
+            Self::Proceed
+        } else {
+            Self::Stop
+        }
+    }
+}
+
+impl From<Propagation> for bool {
+    fn from(c: Propagation) -> Self {
+        match c {
+            Propagation::Stop => false,
```

Comment:
Does this correspond to `Inhibit(true)`?  Then values are flipped.

Reply:
Thanks! Fixed